### PR TITLE
XML parse breaking change in 4.5.2

### LINKED ISSUE #21710: Breaking changes to XML parsers between .NET 4.5.1 and .NET 4.5.2 not outlined
This page https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/runtime/4.0-4.8 does not mention the fact that XmlDocument, XmlReader and perhaps some other classes have a breaking change between 4.5.1 and 4.5.2 - the default XmlUrlResolver is no longer used by default as explained here https://docs.microsoft.com/en-us/dotnet/api/system.xml.xmlreader.create?view=netframework-4.7.2 (search for "4.5.2"). So XML parsing changes big time for certain scenarios. This is a breaking change which is not outlined.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c07873b5-0b45-90dd-fb98-035241a2bed7
* Version Independent ID: 35e4c4f6-fd40-de30-f82e-7a9566ffb90e
* Content: [Runtime Changes for Migration from .NET Framework 4.0 to 4.8](https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/runtime/4.0-4.8)
* Content Source: [docs/framework/migration-guide/runtime/4.0-4.8.md](https://github.com/dotnet/docs/blob/master/docs/framework/migration-guide/runtime/4.0-4.8.md)
* Product: **dotnet-framework**
* Technology: **dotnet-appcompat**
* GitHub Login: @chlowell
* Microsoft Alias: **gewarren**

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #21710.

- Adds EnableLegacyXmlSettings element to config docs.
- Adds breaking change in 4.5.2 for XmlReaderSettings.

Preview links:

- [EnableLegacyXmlSettings](https://review.docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/enablelegacyxmlsettings-element?branch=pr-en-us-21876).
- [XML breaking change](https://review.docs.microsoft.com/en-us/dotnet/framework/migration-guide/runtime/4.5.1-4.5.2?branch=pr-en-us-21876#xml-parsing-changes)

### CODE REVIEW

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

You can :shipit: when ready.

**Comment by gewarren** [comment]
I guess Bill merged this by mistake. @tarekgh I'd still appreciate if you could look at the content. It's pretty bare bones at the moment.

**Comment by BillWagner** [comment]
That was my mistake. This PR was bundled with several customer PRs that I reviewed and merged. I hope that doesn't cause too much disruption.

File: docs/framework/configure-apps/file-schema/runtime/enablelegacyxmlsettings-element.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Specifies whether to use XML parsing settings from .NET Framework 4.5.1 and earlier versions.
+
+[**\<configuration>**](../configuration-element.md)\
+&nbsp;&nbsp;[**\<runtime>**](runtime-element.md)\
+&nbsp;&nbsp;&nbsp;&nbsp;**\<EnableLegacyXmlSettings>**
+
+## Syntax
+
+```xml
+<EnableLegacyXmlSettings enabled="0"|"1" />
```
Comment (tarekgh) [inline docs/framework/configure-apps/file-schema/runtime/enablelegacyxmlsettings-element.md:16]: I think the switch has to be set in the registry. 

https://referencesource.microsoft.com/#System.Xml/System/Xml/Core/XmlReaderSettings.cs,747

**Pull request merged**